### ISSUE TITLE
Setting up custom context within the Shortcut Manager breaks table navigation and trigger an `Uncaught TypeError`

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#9529]
### Description

Setting up custom context within the Shortcut Manager breaks table navigation and triggers an `Uncaught TypeError`.

```
handsontable.full.js:133063 Uncaught TypeError: Cannot read properties of undefined (reading 'hasShortcut')
    at handsontable.full.js:133063:24
    at onkeydown (handsontable.full.js:133516:32)
```

 The bug disappears when we enter the edit mode.

### Steps to reproduce

1. go to https://jsfiddle.net/3cmx5usz/15/
2. open the console
3. try to move with key arrows or open the cell editor using <kbd>Enter</kbd> key

**Result**
The error appears in the console and the table does not respond.

![Kapture 2022-05-25 at 11 25 31](https://user-images.githubusercontent.com/10757813/170229454-193f57e5-7a75-4a56-a3c8-2b5880049a24.gif)

To get rid of the issue we need to open the cell editor via double <kbd>LMB</kbd> click

### Demo

https://jsfiddle.net/3cmx5usz/15/ v12

### Your environment
* Handsontable version: 12.0.0
* Browser name and version: Chrome 101
* Operating system: macOS Monterey
